starter-code/src/viking.js

```diff
@@ -1,11 +1,79 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+
+  this.attack = function () {
+    return this.strength;
+  };
+
+  this.receiveDamage = (damage) => {
+    this.health -= damage;
+  };
+}
 
 // Viking
-function Viking() {}
+
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+  this.receiveDamage = (damage) => {
+    this.health -= damage;
+    return this.health > 0 ? this.name + ' has received ' + damage + ' points of damage' : this.name + ' has died in act of combat';
+  };
+  this.battleCry = function () {
+    return 'Odin Owns You All!';
+  };
+}
+Viking.prototype = Object.create(Soldier.prototype);
 
```

Comment:
```
Viking.prototype.constructor = Viking;

```
So from now on we could re-implement method receivedDamage from Soldier:
```
Viking.prototype.receiveDamage = function(damage) { 
         ....
};
```
and battleCry:
```
Viking.prototype.battleCry = function() {
  return "Odin Owns You All!";
};
```